File: src/Functional/Clojure/singleton.clj
(defonce registry (atom {:count 0}))

(defn registry-instance [] registry)

(let [first (registry-instance)
      second (registry-instance)]
  (swap! first update :count inc)
  (println (str "same=" (if (identical? first second) "true" "false")))
  (println (str "count=" (:count @second))))

File: src/Functional/OCaml/singleton.ml
module Registry = struct
  let state = ref 0
  let instance () = state
end

let () =
  let first = Registry.instance () in
  let second = Registry.instance () in
  incr first;
  Printf.printf "same=%b\n" (first == second);
  Printf.printf "count=%d\n" !second
